### ISSUE TITLE
3.6.2 broken when no SuperTable installed

### ISSUE THREAD

**Issue by vandres** [ethercreative/simplemap#182]
### Description
<img width="1113" alt="Bildschirmfoto 2019-07-23 um 14 00 27" src="https://user-images.githubusercontent.com/1436383/61710666-65e5e700-ad52-11e9-8034-e98534c9010c.png">

### Steps to reproduce

1. Update with no SuperTable plugin
2. Call website

### Additional info

- Craft version: latest
- Maps version: 3.6.2
- PHP version: 7.2.x
- Database driver & version:
- Other Plugins:

**Comment by vandres** [comment]
Possible fix: `$superTableBlockTypes = $pc->get('superTableBlockTypes', true) ?? [];`

**Comment by sebastian-lenz** [comment]
Also fails if there are no matrix blocks defined.

**Comment by MattWilcox** [comment]
+1 to this; had a mild heart attack just when everything everywhere resolved in nothing but this error page. Forntunately a nuking of the /vendor folder, pushing composer.json back to 3.6.1, and running composer install again got me back to a working state of affairs.

**Comment by d--j** [comment]
Also: `$blockType['fields']` does not need to be present (if you have a matrix block with no fields) – guard against that, too.


```php
<?php

namespace ether\simplemap\migrations;

use Craft;
use craft\db\Migration;
use ether\simplemap\fields\MapField;
use yii\base\ErrorException;
use yii\base\Exception;
use yii\base\NotSupportedException;
use yii\web\ServerErrorHttpException;

/**
 * m190723_105637_fix_map_field_column_type migration.
 */
class m190723_105637_fix_map_field_column_type extends Migration
{

    /**
     * @inheritdoc
     * @throws NotSupportedException
     * @throws ErrorException
     * @throws Exception
     * @throws ServerErrorHttpException
     */
    public function safeUp()
    {
        $pc = Craft::$app->getProjectConfig();

        $fields = $pc->get('fields', true) ?? [];
        $matrixBlockTypes = $pc->get('matrixBlockTypes', true) ?? [];
        $superTableBlockTypes = $pc->get('superTableBlockTypes', true) ?? [];
        $updates = [];

        foreach ($fields as $fieldUid => $field)
	        if ($field['type'] === MapField::class)
		        $updates['fields.' . $fieldUid . '.contentColumnType'] = 'text';

        foreach ($matrixBlockTypes as $blockTypeUid => $blockType)
            if (!empty($blockType['fields']))
                foreach ($blockType['fields'] as $fieldUid => $field)
                    if ($field['type'] === MapField::class)
                        $updates['matrixBlockTypes.' . $blockTypeUid . '.fields.' . $fieldUid . '.contentColumnType'] = 'text';



        foreach ($superTableBlockTypes as $blockTypeUid => $blockType)
            if (!empty($blockType['fields']))
                foreach ($blockType['fields'] as $fieldUid => $field)
                    if ($field['type'] === MapField::class)
                        $updates['superTableBlockTypes.' . $blockTypeUid . '.fields.' . $fieldUid . '.contentColumnType'] = 'text';

        foreach ($updates as $path => $value)
            $pc->set($path, $value);
    }

    /**
     * @inheritdoc
     */
    public function safeDown()
    {
        echo "m190723_105637_fix_map_field_column_type cannot be reverted.\n";
        return false;
    }

}

```

**Comment by d--j** [comment]
Hi @Tam 

this is not completely fixed with 5c35311 

The migration also breaks when there are Matrix fields/SuperTable fields that do not have any sub fields (e.g. a "horizontal ruler" matrix block)
